src/main/java/com/example/cats/controller/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,80 @@
+package com.example.cats.controller;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.ConstraintViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+import java.time.Instant;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    record ErrorResponse(int status, String error, String message, String path, String timestamp) {}
```

Comment:
So we have the Problem details in spring, it's out of the box solution please re-watch the version